src/impl/realm_coordinator.cpp

```diff
@@ -330,16 +330,19 @@ void RealmCoordinator::commit_write(Realm& realm)
         // skip version
         std::lock_guard<std::mutex> l(m_notifier_mutex);
 
-        transaction::commit(Realm::Internal::get_shared_group(realm));
+        transaction::commit(*Realm::Internal::get_shared_group(realm));
 
         // Don't need to check m_new_notifiers because those don't skip versions
         bool have_notifiers = std::any_of(m_notifiers.begin(), m_notifiers.end(),
                                           [&](auto&& notifier) { return notifier->is_for_realm(realm); });
         if (have_notifiers) {
-            m_notifier_skip_version = Realm::Internal::get_shared_group(realm).get_version_of_current_transaction();
+            m_notifier_skip_version = Realm::Internal::get_shared_group(realm)->get_version_of_current_transaction();
         }
     }
 
+    // In case the SharedRealm gets descturcted in the did_change callback.
+    auto realm_ref = realm.shared_from_this();
```

Comment:
My comment about this in shared_realm.cpp applies to here as well (`realm` can theoretically be the thing keeping the coordinator alive).